client/src/main/scala/utils/DomHandler.scala

```diff
@@ -70,31 +71,85 @@ object DomHandler {
     _ ← inputs.map(input ⇒ attachKeyUpHandler(input, onkeyup, onEnterPressed)).sequence
   } yield ()
 
+  /** Assigns behavior to an input element change event
+    */
+  def onInputChange(
+    onchange: (String, Seq[String]) ⇒ Coeval[Unit]
+  ): Coeval[Unit] = {
+    for {
+      inputs ← allInputs
+      _ ← inputs.map(input ⇒ attachOnInputChange(input, onchange)).sequence
+      _ ← inputs.map(input ⇒ attachOnInputPaste(input, onchange)).sequence
+    } yield ()
+  }
+
   /** Shows modal for signing up
     */
   def showSignUpModal: Coeval[Unit] = Coeval($("#mustSignUp").modal("show"))
 
+  def updateExerciseInput(
+    input:   HTMLInputElement,
+    actions: ((String, Seq[String])) ⇒ Unit
+  ) = {
+    setInputWidth(input).value
+
+    val maybeInputInfo = for {
+      methodName ← methodParent(input)
+      exercise ← findExerciseByMethod(methodName)
+      inputValues = getInputsValues(exercise)
+    } yield (methodName, inputValues)
+
+    maybeInputInfo.foreach[Unit](actions)
+  }
+
   def attachKeyUpHandler(
     input:          HTMLInputElement,
     onkeyup:        (String, Seq[String]) ⇒ Coeval[Unit],
     onEnterPressed: String ⇒ Coeval[Unit]
   ): Coeval[Unit] = Coeval {
     $(input).keyup((e: dom.KeyboardEvent) ⇒ {
+      updateExerciseInput(
+        input,
+        (info: (String, Seq[String])) ⇒ {
```

Comment:
Let's destructure this tuple in the match below and also give it some good identifiers instead of `._1` `._2`